service/raft/raft_group0_client.cc

```diff
@@ -293,4 +294,36 @@ group0_command raft_group0_client::prepare_command(table_query query) {
     return group0_cmd;
 }
 
+raft_group0_client::query_result_guard::query_result_guard(utils::UUID query_id, raft_group0_client& client)
+    : _query_id{query_id}, _client{&client} {
+    _client->_results.emplace(_query_id, std::nullopt);
```

Comment:
I recommend checking if `emplace` succeeded (i.e. there wasn't an entry with this ID yet) and performing `on_internal_error` if it didn't.